lib/aasm/persistence/active_record_persistence.rb

```diff
@@ -145,7 +145,14 @@ def aasm_ensure_initial_state
         end
 
         def aasm_fire_event(state_machine_name, name, options, *args, &block)
-          success = options[:persist] ? self.class.transaction(:requires_new => requires_new?(state_machine_name)) { super } : super
+          success = if options[:persist] 
+            self.class.transaction(:requires_new => requires_new?(state_machine_name)) do
+              lock!(requires_lock?(state_machine_name)) if requires_lock?(state_machine_name)
```

Comment:
Awesome PR! :smile: :+1: :rainbow: We are facing the same issue on a project right now.

Why `lock!(requires_lock?(state_machine_name))` rather than just `lock!`? 

How could `requires_lock?(state_machine_name)` ever be be passed into `lock!` as `false`?